spec/openid_configuration_spec.rb

```diff
@@ -0,0 +1,42 @@
+require 'spec_helper'
+
+RSpec.describe LoginGov::OidcSinatra::OpenidConfiguration do
+  let(:host) { 'http://localhost:3000' }
+  let(:authorization_endpoint) { "#{host}/openid/authorize" }
+  let(:token_endpoint) { "#{host}/api/openid/token" }
+  let(:jwks_uri) { "#{host}/api/openid/certs" }
+  let(:end_session_endpoint) { "#{host}/openid/logout" }
+  let(:client_id) { 'urn:gov:gsa:openidconnect:sp:sinatra' }
+
+  before do
+    stub_request(:get, "#{host}/.well-known/openid-configuration").
+      to_return(body: {
+        authorization_endpoint: authorization_endpoint,
+        token_endpoint: token_endpoint,
+        jwks_uri: jwks_uri,
+        end_session_endpoint: end_session_endpoint,
+      }.to_json)
+  end
+
+  describe '#live' do
+    it 'raises error if request fails' do
+      stub_request(:get, "#{host}/.well-known/openid-configuration").
+        to_return(body: '', status: 401)
+
+      expect { LoginGov::OidcSinatra::OpenidConfiguration.live }.
+        to raise_error(LoginGov::OidcSinatra::AppError)
+    end
+  end
+
+  describe '#cached' do
+    after(:each) do
+      WebMock.enable!
+    end
+    it 'does not make more than one HTTP request' do
+      oidc_config = LoginGov::OidcSinatra::OpenidConfiguration.cached
+      WebMock.disable!
```

Comment:
the `webmock.disable` doesn't make sense to me? Can we use webmock expectations like `expect(a_request(...)).to have_been_made.once`?

Reply:
Thanks, was not aware of that. It makes things much clearer.